ndc-models/src/relational_query/expression.rs

```diff
@@ -358,6 +358,18 @@ pub enum RelationalExpression {
     Floor {
         expr: Box<RelationalExpression>,
     },
+    /// Only used when in specific contexts where the appropriate capability is supported:
+    /// * During projection: `relational_query.project.expression.scalar.get_field`
+    /// * During filtering: `relational_query.filter.scalar.get_field`
+    /// * During sorting:`relational_query.sort.expression.scalar.get_field`
+    /// * During joining: `relational_query.join.expression.scalar.get_field`
+    /// * During aggregation: `relational_query.aggregate.expression.scalar.get_field`
+    /// * During windowing: `relational_query.window.expression.scalar.get_field`
+    GetField {
+        column: Box<RelationalExpression>,
+        field: RelationalLiteral,
```

Comment:
Would probably be worth introducing a separate type here if we're not going to allow arbitrary expressions, one which only allows `String` or `usize` for object and array indices respectively. No sense in indexing with a date, for example.